src/SpotifyLogin.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from "react"
+import { generateAuthUrl } from "./utils/spotify"
+import axios from "axios";
+
+export const SpotifyLogin = () => {
```

Comment:
Logically there are 2 components here: Login (when user not logged in) and UserInfo (when user logged in). Would be good to split it.

Reply:
I have split SpotifyLogin into 2 components, but now I have to check is user is logged in or not. Now I do it by checking of existing of token in local storage.

Reply:
That's ok to use token as indicator of logged in user.
Start with extracting token related logic into separate file. For example, tokenStorage with getToken/setToken functions. You can use it then directly inside components or pass from the parent components (manually or via  React context https://ru.reactjs.org/docs/context.html)

Reply:
Now I have 2 functions: setToken and getToken. Is it a good approach or should I move those functions into class tokenStorage?

Reply:
Let's start with current implementation and will refactor later if needed.